src/actions.js

```diff
@@ -0,0 +1,55 @@
+export function togglePlayer () {
+  return {
+    type: 'TOGGLE_PLAYER'
+  };
+}
+
+export function incrementStep () {
+  return {
+    type: 'INCREMENT_STEP'
+  };
+}
+
+export function updateBoard (position) {
+  return {
+    type: 'UPDATE_BOARD',
+    position
+  };
+}
+
+const initialState = {
+  history: [{
+    board: Array(9).fill(null)
+  }],
+  stepNumber: 0,
+  xIsNext: true
+};
+
+export const ticTac = (state = initialState, action) => {
+  switch (action.type) {
+    case 'INCREMENT_STEP':
+      return {
+        ...state,
+        stepNumber: state.stepNumber + 1
+      };
+    case 'TOGGLE_PLAYER':
+      return {
+        ...state,
+        xIsNext: !state.xIsNext
+      };
+    case 'UPDATE_BOARD':
+      const position = action.position;
+      const currentPlayer = state.xIsNext ? 'X' : 'O';
+      const oldBoard = state.history[state.history.length - 1].board.slice();
+      oldBoard[position] = currentPlayer;
+      const newBoard = oldBoard;
+      const newHistory = state.history.concat([{ board: newBoard }]);
```

Comment:
By using the spread operator, you should be able to reduce.... ha get it? this reducer down to 1 or 2 lines